e2e/cypress/integration/integrations/custom_slash_commands/custom_slash_commands_spec.js

```diff
@@ -0,0 +1,257 @@
+        cy.visit(`/${team1.name}/channels/town-square`);
+        cy.wait(TIMEOUTS.TWO_SEC);
+
+        // # Run slash command
+        cy.get('#post_textbox', {timeout: TIMEOUTS.ONE_MIN}).should('be.visible').clear().type(`/${trigger}{enter}`);
+        cy.wait(TIMEOUTS.TWO_SEC);
+
+        // * Verify that last post is by newname
+        cy.getLastPost().within(() => {
+            cy.get('.post__header').find('.user-popover').as('usernameForPopover').should('have.text', 'newname');
+        });
+
+        // # Cleanup command
+        deleteCommand(team1, trigger);
+    });
+
+    it('MM-T701 Slash command - Override profile picture', () => {
+        cy.apiUpdateConfig({
+            ServiceSettings: {
+                EnablePostIconOverride: true,
+            },
+        });
+
+        // # Open slash command page
+        cy.visit(`/${team1.name}/integrations/commands/installed`);
+
+        // # Create new Slash command
+        addNewCommand(team1, trigger, commandURL);
+
+        // # Open slash command page
+        cy.visit(`/${team1.name}/integrations/commands/installed`);
+
+        // # Update icon URL
+        cy.get(':nth-child(1) > .item-details > .d-flex > .item-actions > a > span').click();
+        const iconURL = 'http://www.mattermost.org/wp-content/uploads/2016/04/icon_WS.png';
+        cy.get('#iconUrl').type(iconURL);
+        cy.get('#saveCommand').click();
+
+        // # Go back to home channel
+        cy.visit(`/${team1.name}/channels/town-square`);
+        cy.wait(TIMEOUTS.TWO_SEC);
+
+        // # Run slash command
+        cy.get('#post_textbox', {timeout: TIMEOUTS.ONE_MIN}).should('be.visible').clear().type(`/${trigger}{enter}`);
+        cy.wait(TIMEOUTS.TWO_SEC);
+
+        // * Verify that last post has correct icon
+        cy.getLastPost().within(() => {
+            const baseUrl = Cypress.config('baseUrl');
+            const encodedIconUrl = encodeURIComponent(iconURL);
+            cy.get('.profile-icon > img').as('profileIconForPopover').should('have.attr', 'src', `${baseUrl}/api/v4/image?url=${encodedIconUrl}`);
+        });
+
+        // # Cleanup command
+        deleteCommand(team1, trigger);
+    });
+
+    it('MM-T703/MM-704 Show custom slash command in autocomplete', () => {
+        // # Open slash command page
+        cy.visit(`/${team1.name}/integrations/commands/installed`);
+
+        // # Create new Slash command
+        addNewCommand(team1, trigger, commandURL);
+
+        // # Open slash command page
+        cy.visit(`/${team1.name}/integrations/commands/installed`);
+
+        // # Update autocomplete
+        cy.get(':nth-child(1) > .item-details > .d-flex > .item-actions > a > span').click();
+        cy.get('#autocomplete').click();
+        const hint = '[test-hint]';
+        cy.get('#autocompleteHint').type(hint);
+        const desc = 'Auto description';
+        cy.get(':nth-child(10) > .col-md-5 > #description').type(desc);
+        cy.get('#saveCommand').click();
```

Comment:
The `cy.get(':nth-child(x) ...` seem quite "obscure". Is there any better way to identify the elements you are addressing? If not, could you add comments on what exactly are you doing? (e.g. `// # Click the link on the first child, that must be X`).